player-vs-computer.js

```diff
@@ -0,0 +1,240 @@
+const readline = require("readline");
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+function ask(questionText) {
+  return new Promise((resolve, reject) => {
+    rl.question(questionText, resolve);
+  });
+};
+
+//create function for computer guess
+function binarySearch(min, max) {
+  return Math.floor((max - min) / 2 + min);
+};
+
+//create function to generate random number
+function randomNum(min, max) {
+  return Math.floor(Math.random() * (max - min + 1) + min);
+};
+
+start();
+
+// function containing all game logic
+async function start() {
+  console.log(
+    "\n \nLet's play a game.\n \nWe both will pick a random number between two numbers of your choosing, and will take turns trying to guess each other's number.\n \nAfter each guess I will tell you whether your guess is correct, and if not whether my number is higher or lower. You will do the same for me. Some ground rules:\n\n1. The numbers must be integers.\n\n2. The numbers must be positive.\n\n3. It should go without saying, but the top number of the range must be higher than the bottom number of the range.\n\nI WILL catch you if you try to cheat!\n"
+  );
+
+  let name = await ask(
+    "Now, before we begin, would you tell me your name, please?\n\n"
+  );
+
+  console.log(`\nThanks. Nice to meet you, ${name}.\n`);
+  // establish foundational variables
+
+  let rangeBottom;
+  let rangeTop;
+  let playerNum;
+
+  await getBottomRange();
+
+  //get range bottom
+  async function getBottomRange() {
+    let rangeBottomInput = await ask(
+      "To begin, what is the bottom number of the range from which we will select our numbers?\n\n"
+    );
+    rangeBottom = parseInt(rangeBottomInput);
+    await checkBottomRange(rangeBottomInput);
+    return;
+  };
+
+  //make sure range bottom is valid
+  async function checkBottomRange(input) {
+    if (!isNumeric(input)) {
+      console.log(
+        `\n${input} is not a valid number. Enter something else please!\n`
+      );
+      await getBottomRange();
+    } else if (isNumeric(input) === true && parseFloat(input) % 1 != 0) {
+      console.log("\nIntegers only, please!\n");
+      await getBottomRange();
+    } else if (isNumeric(input) === true && parseFloat(input) < 0) {
+      console.log("\nNo negative numbers, please. Let's try again.\n");
+      await getBottomRange();
+    } else {
+      return;
+    };
+  };
```

Comment:
This function is very similar to the function getTopRange. It could be very possible to refactor the two of these functions to be one to save yourself lines of code.